webrtc/RTCTransport-helper.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+/*
+  Code using this helper should also include RTCPeerConnection-helper.js
+  in the main HTML file
+
+  The following helper functions are called from RTCPeerConnection-helper.js:
+    getTrackFromUserMedia
+    addTrackOrTransceiver
+    exchangeIceCandidates
+    doSignalingHandshake
+*/
+
+// Wait for all DTLS and ICE transport of a peer connection
+// to turn to connected or completed state. When the RTCPeerConnectionState
+// is "connected", all RTCIceTransports and RTCDtlsTransports are in the
+// "connected", "completed" or "closed" state and at least one of them is in the
+// "connected" or "completed" state.
+function waitConnectingPc(pc) {
+  return new Promise((resolve, reject) => {
+    if (pc.connectionState === 'connected') {
+      resolve();
+    }
+
+    pc.addEventListener('connectionstatechange', () => {
+      const { connectionState } = pc;
+
+      if (connectionState === 'connected') {
+        resolve();
+      } else if (['closed', 'failed'].includes(connectionState)) {
+        reject(new Error(`one of DTLS/ICE transports transition to unexpected state ${connectionState}`));
+      }
+    });
+  });
+}
+
+function getDtlsTransportFromSctpTransport(sctpTransport) {
+  assert_true(sctpTransport instanceof RTCSctpTransport,
+    'Expect pc.sctp to be instantiated from RTCSctpTransport');
+
+  const dtlsTransport = sctpTransport.transport;
+  assert_true(dtlsTransport instanceof RTCDtlsTransport,
+    'Expect sctp.transport to be an RTCDtlsTransport');
+
+  return dtlsTransport;
+}
+
+function getIceTransportFromSctpTransport(sctpTransport) {
+  const dtlsTransport = getDtlsTransportFromSctpTransport(sctpTransport);
+
+  const iceTransport = dtlsTransport.transport;
+  assert_true(iceTransport instanceof RTCIceTransport,
+    'Expect dtlsTransport.transport to be an RTCIceTransport');
+
+  return iceTransport;
+}
+
+// Get one or more RTCDtlsTransports from RTCRtpSender/Receiver
+// in pc. There may be more than one e.g. there are more than one
```

Comment:
I think this is not quite right. "There may be more than one DTLSTransport if there are multiple underlying transports, such as when not using max-bundle, or when not using RTCP multiplexing."